README.md

```diff
@@ -4,33 +4,301 @@ sktm - skt manager
+credentials like this:
+
+    sudo -u jenkins kinit <YOUR_PRINCIPAL>
+
+Where `<YOUR_PRINCIPAL>` would be the principal you usually login with. E.g.:
+
+    sudo -u jenkins kinit user@EXAMPLE.COM
+
+Note that in this case you would need to periodically refresh the credentials
+Jenkins has as they expire, by re-running the command above.
+
+Make sure you get a dedicated principal and a keytab in production
+environment, instead of letting Jenkins use your credentials.
+
+### Jenkins
+
+To configure Jenkins, open [http://localhost:8080](http://localhost:8080) in
+your browser and follow the wizard. Installing the suggested plugins should be
+sufficient. The examples below assume that the admin user, created during
+Jenkins configuration, was named "sktm", and assigned password "sesame".
+
+At the moment, sktm doesn't support CSRF protection when authenticating to
+Jenkins. So, after the Jenkins setup is complete, navigate to [Manage Jenkins
+-> Configure Global Security](http://localhost:8080/configureSecurity/) and
+uncheck the "Prevent Cross Site Request Forgery exploits" checkbox. Note that
+you shouldn't do this on production, or exposed systems.
+
+Create a configuration file for Jenkins Job Builder, describing how to access
+Jenkins:
+
+    mkdir -p ~/.config/jenkins_jobs
+    cat >~/.config/jenkins_jobs/jenkins_jobs.ini <<"EOF"
+    [jenkins]
+    url=http://localhost:8080
+    user=sktm
+    password=
+    EOF
+
+Then go to ["sktm" user preferences in
+Jenkins](http://localhost:8080/user/sktm/configure), click "Show API
+token...", copy the text from the displayed "API Token" field, and paste it
+into the `password` field value of the configuration file created above. Note:
+secure your configuration file appropriately when configuring production
+systems.
+
+Test that Jenkins Job Builder can reach and authenticate to Jenkins by running
+`jenkins-jobs list`. It should output something like `INFO:root:Matching jobs:
+0` and complete succesfully.
+
+The Jenkins project description ([example-project.yaml](example-project.yaml))
+used in this documentation will run an HTTP server on port 4040 to let Beaker
+download the tested kernel images. If your Jenkins host is behind NAT (e.g. in
+a VM), make sure to setup port redirection, to have it accessible both
+from the Beaker instance and from inside the host, at the same address, and
+change the URL skt will be publishing with:
+
+    sed -ie 's/`hostname -f`:4040/<HOSTNAME>:<PORT>/' example-project.yaml
+
+Here, `<HOSTNAME>` would be the name of the NAT host Beaker would need to
+access, and `<PORT>` would be the redirected port number. E.g.:
+
+    sed -ie 's/`hostname -f`:4040/pc.example.com:4040/' example-project.yaml
+
+You will need to update the file again, if your NAT host name changes.
+
+Create a Jenkins project for sktm to trigger, from the
+[`example-project.yaml`](example-project.yaml) file, using Jenkins Job
+Builder:
+
+    jenkins-jobs update example-project.yaml
```

Comment:
I'm getting `jenkins.JenkinsException: Error in request. Possibly authentication failed [500]: Server Error` here. The list command from above completes successfully. The only difference in the setup is that my user is still called `admin` but I've reflected on this in the config file. Have you encountered this error?

Reply:
Nope. Check with `strace` if `jenkins-jobs` actually reads its configuration file where credentials are specified. Recheck if credentials are correct. Cannot give any better advice at the moment, sorry.

Reply:
Despite the error message, the problem doesn't seem auth-related. The credentials are read and correct. I'll try to figure out what the issue is but I've never touched jenkins so I can't promise anything

Reply:
Argh. I would like to cater this HOWTO to F28, and it would be great to have this resolved and described. Do you have any news on this?